cerebrium/cortex/advanced-functionality/faster-model-loading.mdx

```diff
@@ -0,0 +1,282 @@
+        model = deserialise_saved_model(model_path, model_id, plaid=faster)
+    else:
+        # some function where you load your model, send it to GPU and prep it for inference
+        model = load_weights_func()
+        print((f"Serialising model to {model_path}"),  file=sys.stderr)
+        serialise_model(model, model_path)
+    return model
+
+
+def deserialise_saved_model(model_path, model_id, plaid=True):
+    """Deserialise the model from the model_path and load into GPU memory"""
+    from tensorizer import TensorDeserializer
+    from tensorizer.utils import no_init_or_tensor
+
+
+    # create a config object that we can use to init an empty model
+    config = AutoConfig.from_pretrained(model_id)
+
+    # Init an empty model without loading weights into gpu. We'll load later.
+    print(("Initialising empty model"),  file=sys.stderr)
+    start = time.time()
+    with no_init_or_tensor():
+        # Load your model here using whatever class you need to initialise an empty model from a config.
+        model = AutoModelForCausalLM.from_config(config)
+    end_init = time.time() - start
+
+    # Create the deserialiser object
+    #   Note: plaid_mode is a flag that does a much faster deserialisation but isn't safe for training.
+    #    -> only use it for inference.
+    deserializer = TensorDeserializer(model_path, plaid_mode=True)
+
+    # Deserialise the model straight into GPU (zero-copy)
+    print(("Loading model"),  file=sys.stderr)
+    start = time.time()
+    deserializer.load_into_module(model)
+    end = time.time()
+    deserializer.close()
+
+    # Report on the timings.
+    print(f"Initialising empty model took {end_init} seconds",  file=sys.stderr)
+    print((f"\nDeserialising model took {end - start} seconds\n"),  file=sys.stderr)
+
+    return model
+
+
+def serialise_model(model, model_path):
+    """Serialise the model and saving the weights to the model_path"""
+    from tensorizer import TensorSerializer
+    try:
+        serializer = TensorSerializer(model_path)
+        start = time.time()
+        serializer.write_module(model)
+        end = time.time()
+        print((f"Serialising model took {end - start} seconds"),  file=sys.stderr)
+        serializer.close()
+        return True
+    except Exception as e:
+        print("Serialisation failed with error: ", e,  file=sys.stderr)
+        return False
+```
+
+## Pytorch Lightning
+
+[Pytorch Lightning's fabric](https://lightning.ai/docs/fabric/latest/) is a library that utilises the optimisations in _Pytorch Lightning_ but at higher level.  
+One of the features of **fabric** is that it allows you to load your model from storage into GPU memory in a single step using their serialised checkpoints.
+
+In our testing, we observed that this method is slightly slower than using **Tensorizer** but it is still faster than huggingface's default method of loading the model by around 10%.
+
```

Comment:
Also maybe just worth mentioning Lighning and putting in a different github repo